sdk/endpoints/users/v2/endpoints.go

```diff
@@ -0,0 +1,67 @@
+package v2
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	endpointsHelpers "github.com/nobl9/nobl9-go/internal/endpoints"
+)
+
+const (
+	baseAPIPath = "usrmgmt/v2/users"
+)
+
+//go:generate ../../../../bin/ifacemaker -y " " -f ./*.go -s endpoints -i Endpoints -o endpoints_interface.go -p "$GOPACKAGE"
+
+func NewEndpoints(client endpointsHelpers.Client) Endpoints {
+	return endpoints{client: client}
+}
+
+type endpoints struct {
+	client endpointsHelpers.Client
+}
+
+// GetUsers fetches a list of [User] filtered by the provided search phrase.
+func (e endpoints) GetUsers(ctx context.Context, params GetUsersRequest) ([]*User, error) {
```

Comment:
I recommend implementing pagination support from the start. Without it, larger organizations may be unable to retrieve their full user list, which could significantly limit the usability of this feature. Planning for scalability now can save us from rework and technical debt later.

Reply:
Good point! As discussed, let's target this with another task https://github.com/nobl9/nobl9-go/issues/727.